### ISSUE TITLE
Getting multiple error messages while complete() call to acknowledge messages

### ISSUE THREAD

**Issue by dakshme** [Azure/azure-sdk-for-java#32160]
### Library name and version

<groupId>com.azure</groupId><artifactId>azure-messaging-servicebus</artifactId><version>7.12.1</version>

### Query/Question

I am using java azure sdk with below dependency details (I was redirected by azure support team to report library specific queries here):
<groupId>com.azure</groupId>
<artifactId>azure-messaging-servicebus</artifactId>
<version>7.12.1</version>

Query:
I am using the above lib to read messages from the azure service bus. Application is reading a chunk of 1000 messages and after processing (which takes about few seconds) it starts marking complete() to these messages. While doing so I see the below error generated by library. I do not understand the reason for such error and need help to understand then and fix them (in case I need to make any change in my implementation)

The link 'recordstopic/subscriptions/wke-listener_23ea1e_1668452898772' is force detached. Code: aggregate-link16969697. Details: AmqpMessagePartitioningEntityMessageCache.IdleTimerExpired: Idle timeout in Seconds: 900., errorContext[NAMESPACE: xxxxxxxxx-bus.servicebus.windows.net. ERROR CONTEXT: N/A, PATH: recordstopic/subscriptions/some-listener, REFERENCE_ID: 7039C12600D94BEC0102EFE163738E7B_G36, LINK_CREDIT: 0]
linkName

The lock supplied is invalid. Either the lock expired, or the message has already been removed from the queue. For more information please see https://aka.ms/ServiceBusExceptions . Reference:a4de9f39-c7cc-4df3-8ab9-fae308c61450, TrackingId:11c2076a-0347-4b48-9bc5-58d7b419ae8e_B1, SystemTracker:xxxxxxxx:topic:recordstopic~63|some-listener, Timestamp:2022-11-15T13:26:17

I see multiple occurrences of such error messages constantly in application logs. I need to understand the root cause for the same.

### Environment

Application is written in java 8. Running in AKS.

**Comment by ki1729** [comment]
@dakshme Can you please share your client configuration code? 
One thing to watch out for is this known issue with AutoComplete - https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/servicebus/azure-messaging-servicebus/TROUBLESHOOTING.md#autocomplete-issue

If you are not already doing this, I would recommend calling `.disableAutoComplete()` on your client initalization and calling message.complete() explicity after you finish processing your message.

**Comment by dakshme** [comment]
Hi @ki1729 Thanks for you reply. Please find the below code I am using to create receiver client. I am using sync client and make a batch read call.
            
```
ServiceBusReceiverClient receiver = new ServiceBusClientBuilder()
                    .connectionString(sbTopicConnUri)
                    .receiver()
                    .topicName(azureConfig.getSbTopicName())
                    .subscriptionName(azureConfig.getSbTopicSubscription())
                    .disableAutoComplete()
                    .buildClient();
```
Using above client application making reach calls like this:

`messageReceiver.receiveMessages(1000, Duration.ofSeconds(10));`

It returns the messages but intermittently we gets the mentioned error emitted by library. In certain environments the frequency is high of getting such error logs.

Also I notice one more issue while using the above receiver client with batch receive messages. Many times the receiver call shows inconsistency in terms of returning number of msgs from subscription. Specifically when the number of available messages are low on subscription (say under 50). In this case I see messages received/returned by library is much less like  1 or 2 and many time it even returns 0 messages for long time even after polling multiple times. Which is causing problem that it take very long time to process certain messages. Please note batch receiver seems to be working well when there is bulk or load of messages on the subscription.
Do u think batch read 1000 msgs with  timeout 10 sec may cause this behavior ? If yes, do u suggests any kind of optimization or configuration to optimize it.

so basically 2 issues we are facing, error logs and receive batch message inconsistency (when low msgs available in subscription). The later one is more burning for us now.

**Comment by dakshme** [comment]
Recently I have noticed another behavior impacting the execution of the service. I see the below exception raised by lib (one in a while in many days):
```
2022-12-11 07:18:33.294 ERROR wke-scheduler-5c6fbbb48d-pjr76 --- [     parallel-1] r.c.p.Operators                          correlation-id= data-partition-id=: Operator called default onErrorDropped
reactor.core.publisher.Sinks$EmissionException: Spec. Rule 1.3 - onSubscribe, onNext, onError and onComplete signaled to a Subscriber MUST be signaled serially.
    at reactor.core.publisher.InternalManySink.emitComplete(InternalManySink.java:86)
    at com.azure.messaging.servicebus.SynchronousReceiveWork.complete(SynchronousReceiveWork.java:205)
    at com.azure.messaging.servicebus.SynchronousReceiveWork.complete(SynchronousReceiveWork.java:179)
    at com.azure.messaging.servicebus.SynchronousReceiveWork.lambda$start$3(SynchronousReceiveWork.java:120)
    at reactor.core.publisher.LambdaSubscriber.onNext(LambdaSubscriber.java:160)
    at reactor.core.publisher.FluxSwitchMap$SwitchMapMain.drain(FluxSwitchMap.java:355)
    at reactor.core.publisher.FluxSwitchMap$SwitchMapMain.innerNext(FluxSwitchMap.java:413)
    at reactor.core.publisher.FluxSwitchMap$SwitchMapInner.onNext(FluxSwitchMap.java:512)
    at reactor.core.publisher.MonoDelay$MonoDelayRunnable.propagateDelay(MonoDelay.java:271)
    at reactor.core.publisher.MonoDelay$MonoDelayRunnable.run(MonoDelay.java:286)
    at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
    at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```

After this exception the scheduler thread in the application suddenly stops working. Due to which processing of the further messages stops. I am not sure how this error/exception (if unhandled) impacting the scheduler thread of the application. I found a github issues (https://github.com/reactor/reactor-core/issues/2579) around this. Could this be anyways related to my issue.

Please suggest us here as this intermittent behavior is  blocking our production env.
